src/engine/controls/cuecontrol.cpp

```diff
@@ -1801,30 +2100,138 @@ void CueControl::hotcueFocusColorNext(double v) {
     pCue->setColor(colorPalette.nextColor(*color));
 }
 
+void CueControl::setCurrentSavedLoopControl(HotcueControl* pControl) {
+    if (m_pCurrentSavedLoopControl && m_pCurrentSavedLoopControl != pControl) {
+        // Disable previous saved loop
+        DEBUG_ASSERT(m_pCurrentSavedLoopControl->getStatus() != HotcueControl::Status::Invalid);
+        m_pCurrentSavedLoopControl->setStatus(HotcueControl::Status::Valid);
+        m_pCurrentSavedLoopControl = nullptr;
+    }
+
+    if (!pControl) {
+        return;
+    }
+
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack) {
+        return;
+    }
+
+    CuePointer pCue(pControl->getCue());
+
+    // Need to unlock before emitting any signals to prevent deadlock.
+    lock.unlock();
+
+    VERIFY_OR_DEBUG_ASSERT(pCue &&
+            pCue->getType() == mixxx::CueType::Loop &&
+            pCue->getEndPosition() != Cue::kNoPosition) {
+        return;
+    }
+
+    // Set new control as active
+    m_pCurrentSavedLoopControl = pControl;
+    setLoop(pCue->getPosition(), pCue->getEndPosition(), true);
+    pControl->setStatus(HotcueControl::Status::Active);
+}
+
+void CueControl::slotLoopReset() {
+    setCurrentSavedLoopControl(nullptr);
+}
+
+void CueControl::slotLoopEnabledChanged(bool enabled) {
+    if (!m_pCurrentSavedLoopControl) {
+        return;
+    }
+
+    DEBUG_ASSERT(m_pCurrentSavedLoopControl->getStatus() != HotcueControl::Status::Invalid);
+    DEBUG_ASSERT(
+            m_pCurrentSavedLoopControl->getCue() &&
+            m_pCurrentSavedLoopControl->getCue()->getPosition() ==
+                    m_pLoopStartPosition->get());
+    DEBUG_ASSERT(
```

Comment:
These assertions can trigger is a saved loop is active and the track is unloaded. Gotta look into that.

Reply:
Fixed.

Reply:
I was too quick. This can still happen if the loop in/out buttons are used while a saved loop is running.

Reply:
Done.